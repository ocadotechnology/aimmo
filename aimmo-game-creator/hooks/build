#!/bin/bash
docker build --target runner -t $IMAGE_NAME .
