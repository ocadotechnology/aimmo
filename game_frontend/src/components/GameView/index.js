import styled from 'styled-components'
import React, { Component } from 'react'
import Unity from 'react-unity-webgl'
import PropTypes from 'prop-types'
import { CircularProgress } from '@material-ui/core'
import { unityContent } from 'api/unity'

export const GameViewLayout = styled.div`
  grid-area: game-view;
`

export const LoadingBackgroundOverlay = styled.div`
  height: 100%;
  width: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
  background-color: rgba(0, 0, 0, 0.2);
`

export const StyledUnity = styled(Unity)`
  display: ${props => props.gameDataLoaded ? 'block' : 'none'};
`

export const StyledCircularProgress = styled(CircularProgress)`
    max-width: 50%;
`

export default class GameView extends Component {
  constructor (props) {
    super(props)
    unityContent.on('loaded', this.unityContentLoaded)
  }

  unityContentLoaded = () => {
    this.props.connectToGame()
  }

  renderLoadingScreen = gameDataLoaded => {
    if (!gameDataLoaded) {
      return (
        <LoadingBackgroundOverlay>
          <StyledCircularProgress color='inherit' />
        </LoadingBackgroundOverlay>
      )
    }
  }

  renderUnityView = (unityContent, gameDataLoaded) => {
    return (
      <StyledUnity
        gameDataLoaded={gameDataLoaded}
        unityContent={unityContent}
        height='100%'
        width='100%' />
    )
  }

  render () {
    return (
      <GameViewLayout>
        {this.renderLoadingScreen(this.props.gameDataLoaded)}
        {this.renderUnityView(unityContent, this.props.gameDataLoaded)}
      </GameViewLayout>
    )
  }
}

GameView.propTypes = {
  connectToGame: PropTypes.func,
  gameDataLoaded: PropTypes.bool
}
