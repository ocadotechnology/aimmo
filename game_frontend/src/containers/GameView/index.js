import styled from 'styled-components'
import React, { Component } from 'react'
import Unity from 'react-unity-webgl'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { actions } from 'features/Game'

export const GameViewLayout = styled.div`
  grid-area: game-view
`

const UNITY_LOADED = 1

export class GameView extends Component {
  onProgress (progression) {
    if (progression === UNITY_LOADED) {
      this.props.connectToGame()
    }
  }

  render () {
    return (
      <GameViewLayout>
        <Unity
          src='/static/unity/Build/unity.json'
          loader='/static/unity/Build/UnityLoader.js'
          onProgress={this.onProgress.bind(this)}
        />
      </GameViewLayout>
    )
  }
}

GameView.propTypes = {
  connectToGame: PropTypes.func
}

const mapStateToProps = state => ({})

const mapDispatchToProps = {
  connectToGame: actions.socketConnectToGameRequest
}

export default connect(mapStateToProps, mapDispatchToProps)(GameView)
