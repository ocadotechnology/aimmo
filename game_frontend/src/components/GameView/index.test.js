/* eslint-env jest */
import React from 'react'
import GameView, { GameViewLayout } from 'components/GameView'
import { shallow } from 'enzyme/build/index'

describe('<GameView />', () => {
  it('matches snapshot', () => {
    const component = shallow(<GameView />)
    expect(component).toMatchSnapshot()
  })
})

describe('<GameViewLayout />', () => {
  it('matches snapshot', () => {
    const tree = shallow(<GameViewLayout />)
    expect(tree).toMatchSnapshot()
  })
})
