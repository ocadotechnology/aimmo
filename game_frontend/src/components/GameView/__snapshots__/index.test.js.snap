// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GameView /> does not show the loading bar when the game has loaded 1`] = `
<styled.div>
  <Styled(Unity)
    gameDataLoaded={true}
    height="100%"
    unityContent={
      UnityContent {
        "buildJsonPath": "/static/unity/Build/unity.json",
        "constructor": [MockFunction],
        "on": [MockFunction] {
          "calls": Array [
            Array [
              "loaded",
              [Function],
            ],
            Array [
              "loaded",
              [Function],
            ],
          ],
          "results": Array [
            Object {
              "isThrow": false,
              "value": undefined,
            },
            Object {
              "isThrow": false,
              "value": undefined,
            },
          ],
        },
        "send": [MockFunction],
        "setComponentInstance": [MockFunction],
        "setFullscreen": [MockFunction],
        "setUnityInstance": [MockFunction],
        "triggerUnityEvent": [MockFunction],
        "uniqueID": 1,
        "unityConfig": Object {
          "adjustOnWindowResize": true,
          "id": "nill",
          "modules": Object {},
          "unityVersion": 2,
        },
        "unityEvents": Array [],
        "unityLoaderJsPath": "/static/unity/Build/UnityLoader.js",
      }
    }
    width="100%"
  />
</styled.div>
`;

exports[`<GameView /> shows loading bar whilst game is loading 1`] = `
<styled.div>
  <styled.div>
    <Styled(WithStyles(CircularProgress))
      color="inherit"
    />
  </styled.div>
  <Styled(Unity)
    gameDataLoaded={false}
    height="100%"
    unityContent={
      UnityContent {
        "buildJsonPath": "/static/unity/Build/unity.json",
        "constructor": [MockFunction],
        "on": [MockFunction] {
          "calls": Array [
            Array [
              "loaded",
              [Function],
            ],
          ],
          "results": Array [
            Object {
              "isThrow": false,
              "value": undefined,
            },
          ],
        },
        "send": [MockFunction],
        "setComponentInstance": [MockFunction],
        "setFullscreen": [MockFunction],
        "setUnityInstance": [MockFunction],
        "triggerUnityEvent": [MockFunction],
        "uniqueID": 1,
        "unityConfig": Object {
          "adjustOnWindowResize": true,
          "id": "nill",
          "modules": Object {},
          "unityVersion": 2,
        },
        "unityEvents": Array [],
        "unityLoaderJsPath": "/static/unity/Build/UnityLoader.js",
      }
    }
    width="100%"
  />
</styled.div>
`;

exports[`<GameViewLayout /> matches snapshot 1`] = `
.c0 {
  grid-area: game-view;
}

<div
  className="c0"
/>
`;

exports[`<LoadingBackgroundOverlay /> matches snapshot 1`] = `
.c0 {
  height: 100%;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: rgba(0,0,0,0.2);
}

<div
  className="c0"
/>
`;

exports[`<StyledCircularProgress /> matches snapshot 1`] = `
.c0 {
  max-width: 50%;
}

<WithStyles(CircularProgress)
  className="c0"
/>
`;

exports[`<StyledUnity /> is not shown when game data is loading 1`] = `
.c0 {
  display: none;
}

<Unity
  className="c0"
/>
`;

exports[`<StyledUnity /> is shown when game data has loaded 1`] = `
.c0 {
  display: block;
}

<Unity
  className="c0"
  gameDataLoaded={true}
/>
`;
